src/EnergyPlus/WaterManager.cc

```diff
@@ -951,6 +956,40 @@ namespace WaterManager {
             }
             state.dataWaterData->RainFall.CurrentRate = schedRate * ScaleFactor / DataGlobalConstants::SecInHour; // convert to m/s
             state.dataWaterData->RainFall.CurrentAmount = state.dataWaterData->RainFall.CurrentRate * (TimeStepSys * DataGlobalConstants::SecInHour);
+        } else {
+            // placeholder: add EP checks for out of range precipitation value later -- yujie
+            // when there's no site:precipitation but non-zero epw precipitation, uset the epw precipitation as the CurrentRate
+            if (state.dataEnvrn->LiquidPrecipitation > 0.0) {
```

Comment:
Same comments here as above - expected to see this kind of warning come from the weather manager. But then weather manager needs to know if anyone is using the epw precip amount. But that might be a better solution, just thinking out loud here: If there is a Site:Precipitation object, we don't need to do any checking of the epw rainfall amounts, because Site:Precipitation will override that. If there isn't a Site:Precipitation object, then any place that uses precip from the weather file can set `state.dataEnvrn-->weatherFilePrecipAmountsUsed = true` and the weather manager can track and warn about suspicious rainfall amounts so the other objects don't need to?

Reply:
Agree, this seems more logical. I will modify the code structure and put the warning in the weather manager when Site:Precipitation is not present, so subsequent uses of LiquidPrecipitation will all have this warning.